Behat/MailCatcherContext.php

```diff
@@ -247,14 +300,21 @@ public function getMailCatcherClient()
      *
      * @return Message
      */
-    protected function findMail($type, $value)
+    protected function findMail($type, $value, $negation = false)
```

Comment:
This implementation of negation is complex.

I'd recommend a separated method for this purpose, to keep code simple:

```php
protected function findNoMail($type, $value)
{
    try {
		$this->findMail($type, $value);
    } catch (\InvalidArgumentException $e) {
        // Mail not found is OK
        return;
    }

	throw new \InvalidArgumentException(sprintf('A message corresponding to your criterias was found: "%s".', json_encode($criterias)));
}
```